docs/internal/testing/testing.md

```diff
@@ -0,0 +1,64 @@
+# Import Pathing Guidelines for Spec files
+
+## The Problem
+One of the difficult problems that we deal with as library authors is defining the API that we expose to consuming application developers. This essentially means one of two things:
```

Comment:
To me, the mental transition between these two sentences and into the bullet points is jarring. The first sentence talks about something really concrete, a problem. The bullet points provide possible solutions to the problem. But the sentence in between is pretty vague and I don't think it glues the two together very well. I think "We could handle this problem in one of two ways:" flows better.